### PR TITLE
in setupAndSolveQP, it can use both old kinsol and new kinsol

### PR DESCRIPTION
**Pull request by hongkai-dai**
QP controller can call setupAndSolveQP with both new kinsol and the old one. I tested it in drakeAtlasSimulation in DRC.
@tkoolen @kuindersma  , could you review the change? I only made modification for the kinematics part in the QP controller, I should not need to change the call to `HandC`?

### CODE REVIEW

**Comment by hongkai-dai** [comment]
should solve #910

**Comment by hongkai-dai** [comment]
By using new kinsol in QP controller, we can lift the hand which goes through the hand Euler angle singularity smoothly (pitch=pi/2). Checkout the following video
https://www.youtube.com/watch?v=6jjtoJN70Lo&feature=youtu.be
And this is the video using the old kinsol
https://www.youtube.com/watch?v=aOe4lugqZRk

**Comment by RussTedrake** [comment]
excellent!

**Comment by kuindersma** [comment]
Looks good. I would suggest using the updated unit tests in https://github.com/mitdrc/drc/pull/1227 as a check for this code.

**Comment by RussTedrake** [comment]
should i wait for that check before pulling?

File: systems/controllers/QPCommon.cpp [hunk 1]
```diff
@@ -502,21 +502,24 @@ int setupAndSolveQP(NewQPControllerData *pdata, std::shared_ptr<drake::lcmt_qp_c
   
   // add in body spatial equality constraints
   // VectorXd body_vdot;
-  MatrixXd orig = MatrixXd::Zero(4,1);
-  orig(3,0) = 1;
+  Vector4d orig1 = Vector4d::Zero();
+  orig1(3) = 1.0;
+  Matrix<double,3,Eigen::Dynamic> orig(3,1);
```
Comment (rdeits) [inline systems/controllers/QPCommon.cpp]: why is this of size (3, Dynamic) when its size is known to be (3, 1) at compile time?

File: systems/controllers/QPCommon.cpp [hunk 2]
```diff
@@ -502,21 +502,24 @@ int setupAndSolveQP(NewQPControllerData *pdata, std::shared_ptr<drake::lcmt_qp_c
   
   // add in body spatial equality constraints
   // VectorXd body_vdot;
-  MatrixXd orig = MatrixXd::Zero(4,1);
-  orig(3,0) = 1;
+  Vector4d orig1 = Vector4d::Zero();
+  orig1(3) = 1.0;
```
Comment (rdeits) [inline systems/controllers/QPCommon.cpp]: why do we need `orig` and `orig1`?

File: systems/controllers/QPCommon.cpp (hunk 1 above)
Comment (hongkai-dai) [inline systems/controllers/QPCommon.cpp]: The template for forwardJacDotTimesV is only instantiated for pts being Matrix<double,3,Eigen::Dynamic>, I tried to instantiate forwardJacDotTimesV for Matrix<double,3,1> but got linking error for dcrossProduct. @tkoolen ,  did you encounter this problem before?

File: systems/controllers/QPCommon.cpp (hunk 2 above)
Comment (hongkai-dai) [inline systems/controllers/QPCommon.cpp]: The old kinsol expects pts as 4x1 vector, with the last entry being 1. The new kinsol only uses 3x1 vector, so I need both.

**Comment by hongkai-dai** [comment]
I tested this using the split QP controller already. I can test it again with Robin using his new changes tomorrow.

File: systems/controllers/QPCommon.cpp [hunk 3]
```diff
@@ -367,7 +367,7 @@ int setupAndSolveQP(NewQPControllerData *pdata, std::shared_ptr<drake::lcmt_qp_c
 
   MatrixXd R_DQyD_ls = R_ls + D_ls.transpose()*Qy*D_ls;
 
-  pdata->r->doKinematics(robot_state.q,false,robot_state.qd);
+  pdata->r->doKinematics(robot_state.q,true,robot_state.qd);
```
Comment (tkoolen) [inline systems/controllers/QPCommon.cpp]: Why do we need second derivatives?

File: systems/controllers/QPCommon.cpp (hunk 1 above)
Comment (tkoolen) [inline systems/controllers/QPCommon.cpp]: We just discussed this in person.

File: systems/controllers/QPCommon.cpp (hunk 2 above)
Comment (tkoolen) [inline systems/controllers/QPCommon.cpp]: Using orig.colwise().homogeneous() might be cleaner.

**Comment by tkoolen** [comment]
Looks great! Excited to move towards the new stuff!

File: systems/controllers/QPCommon.cpp (hunk 3 above)
Comment (hongkai-dai) [inline systems/controllers/QPCommon.cpp]: forwardJacDotTimesV expects this flag to be true, otherwise it will throw an error.

Comment (hongkai-dai) [inline systems/controllers/QPCommon.cpp]: Sorry It is not forwardJacDotTimesV, but getCOMJacDot. Is there a function that can return Jcom_dot*v directly without calling getCOMJacDot?

Comment (tkoolen) [inline systems/controllers/QPCommon.cpp]: Yep, centerOfMassJacobianDotTimesV.

Comment (hongkai-dai) [inline systems/controllers/QPCommon.cpp]: @kuindersma , in the QP controller, it computes Jcomdot by calling getCOMJacDot and stores it in the pdata. Can we remove the Jdot stored in pdata? I do not think we really need Jdot in this case. What we really need is JdotTimesV, and it is implemented in new doKinematics without calling the second order gradient.

Comment (kuindersma) [inline systems/controllers/QPCommon.cpp]: fine with me

**Comment by RussTedrake** [comment]
@hongkai-dai - still waiting for your code update on this one.

**Comment by hongkai-dai** [comment]
I will reopen a pull request, this one contains the commits that has been merged already.

**Pull request closed without merging**